pkg/odo/cli/component/common_push.go

```diff
@@ -330,3 +332,75 @@ func filterIgnores(filesChanged, filesDeleted, absIgnoreRules []string) (filesCh
 	}
 	return filesChangedFiltered, filesDeletedFiltered
 }
+
+// retrieveDefaultComponentNamespace tries to retrieve the current active namespace
+// to set as a default namespace
+func retrieveKubernetesDefaultNamespace() (string, error) {
+	// Get current active namespace
+	client, err := kclient.New()
+	if err != nil {
+		return "", err
+	}
+	return client.Namespace, nil
+}
+
+// checkNamespaceDefaultIfNotAvailable checks to see if a namespace has been passed in, if not, we will default
+// to the standard kubernetes namespace.
+func checkCmdNamespaceDefaultIfNotAvailable(cmd *cobra.Command) (string, error) {
+	var componentNamespace string
+	var err error
+
+	// For "odo create" check to see if --project has been passed.
+	if cmd.Flags().Changed("project") {
+		componentNamespace, err = cmd.Flags().GetString("project")
+		if err != nil {
+			return "", err
+		}
+	} else if cmd.Flags().Changed("namespace") {
+		// For "odo push" check to see if project has been passed
+		componentNamespace, err = cmd.Flags().GetString("namespace")
+		if err != nil {
+			return "", err
+		}
+	} else {
+		componentNamespace, err = retrieveKubernetesDefaultNamespace()
+		if err != nil {
+			return "", err
+		}
+	}
+
+	return componentNamespace, nil
+}
+
+// gatherName parses the Devfile retrieves an appropriate name in two ways.
+// 1. If metadata.name exists, we use it
+// 2. If metadata.name does NOT exist, we use the folder name where the devfile.yaml is located
+func gatherName(devfilePath string) (string, error) {
+
+	// Parse the Devfile
+	devObj, err := devfile.Parse(devfilePath)
+	if err != nil {
+		return "", errors.Wrap(err, "unable to parse devfile while gathering the component name")
+	}
+	metadata := devObj.Data.GetMetadata()
+
+	klog.V(4).Infof("metadata.Name: %s", metadata.Name)
+
+	// 1. Use metadata.name if it exists
+	if metadata.Name != "" {
+
+		// Make sure that we remove any "dashes" in the name as they will not work
```

Comment:
I don't think so. As having the name as "foobar-" is valid within the JSON schema. So doesn't make sense to implement ValidateName in here since technically this is all valid.

Reply:
The issue, though, that ultimately this name will be used in a k8s cluster, won't it? So shouldn't it be valid in that respect? If not, the comment should at least be changed since it doesn't reflect what the function does.